spec/acceptance/suites/default/01_connection_spec.rb

```diff
@@ -22,31 +22,41 @@
       EOF
     }
     let(:domain) { fact_on(host, 'domain') }
-    let(:minion_stunnel_conf) { <<-EOF
-        debug = err
-        syslog = yes
-        pid = /var/run/stunnel.pid
-        engine = auto
-        fips = no
-        foreground = yes
-        [nfs]
-        connect = 10.0.71.106:2049
-        accept = 127.0.0.1:20490
-        client = yes
-        failover = rr
-        key = /etc/pki/simp_apps/stunnel/x509/private/#{host}.#{domain}.pem
-        cert = /etc/pki/simp_apps/stunnel/x509/public/#{host}.#{domain}.pub
-        CAfile = /etc/pki/simp_apps/stunnel/x509/cacerts/cacerts.pem
-        CRLpath = /etc/pki/simp_apps/stunnel/x509/crl
-        ciphers = HIGH:-SSLv2
-        verify = 2
-        delay = no
-        retry = no
-        renegotiation = yes
-        reset = yes
-      EOF
-    }
 
+    # context 'with an existing stunnel process' do
+    #   let(:minion_stunnel_conf) { <<-EOF
+    #       debug = err
+    #       syslog = yes
+    #       pid = /var/run/stunnel.pid
+    #       engine = auto
+    #       fips = no
+    #       foreground = yes
+    #       [nfs]
+    #       connect = 10.0.71.106:2049
+    #       accept = 127.0.0.1:20490
+    #       client = yes
+    #       failover = rr
+    #       key = /etc/pki/simp-testing/pki/private/#{host}.#{domain}.pem
+    #       cert = /etc/pki/simp-testing/pki/public/#{host}.#{domain}.pub
+    #       CAfile = /etc/pki/simp-testing/pki/cacerts/cacerts.pem
+    #       CRLpath = /etc/pki/simp-testing/pki/crl
+    #       ciphers = HIGH:-SSLv2
+    #       verify = 2
+    #       delay = no
+    #       retry = no
+    #       renegotiation = yes
+    #       reset = yes
+    #     EOF
+    #   }
+    #   it 'should set up a stunnel process, ripe for killing' do
+    #     create_remote_file(host, '/etc/stunnel/stunnel.conf', minion_stunnel_conf)
+    #     on(host, 'nohup stunnel /etc/stunnel/stunnel.conf &')
```

Comment:
It hangs on this, which makes sense, but I would rather get this fix through than to figure out how to get this stunnel to run in the background.

Reply:
You can spawn a process in a thread.

Reply:
https://github.com/simp/pupmod-simp-libreswan/blob/master/spec/acceptance/suites/default/00_default_spec.rb#L195